src/renderer/domains/collectives/referendum/resource.ts

```diff
@@ -103,35 +133,43 @@ async function mapReferendum(
         tally: info.data.tally,
         decisionDeposit: info.data.decisionDeposit,
         submissionDeposit: info.data.submissionDeposit,
+        proposal,
+        ends,
       };
+    }
     case 'Approved':
     case 'Rejected':
     case 'Cancelled':
-    case 'TimedOut':
+    case 'TimedOut': {
       return {
         id,
         type: info.type,
         since: info.data.since,
         submissionDeposit: info.data.submissionDeposit,
         decisionDeposit: info.data.decisionDeposit,
       };
-    case 'Killed':
+    }
+    case 'Killed': {
       return {
         id,
         type: info.type,
         since: info.data,
       };
+    }
   }
 }
 
 export async function mapReferendums(
   list: { id: ReferendumId; info: ReferendaReferendumInfoConvictionVotingTally | null }[],
   api: ApiPromise,
+  palletType: CollectivePalletsType,
 ) {
+  const undecidingTimeout = referendaPallet.consts.undecidingTimeout(palletType, api);
+  const tracks = referendaPallet.consts.tracks(palletType, api);
   const value: Referendum[] = [];
```

Comment:
English is poor in words but we still have some 😉
```suggestion
  const referendums: Referendum[] = [];
```